components/mailbox/src/Mailbox.svelte

```diff
@@ -185,9 +209,115 @@
   }
+
+  async function threadStarred(event: CustomEvent) {
+    if (starredThreads.has(event.detail.thread)) {
+      starredThreads.delete(event.detail.thread);
+    } else {
+      starredThreads.add(event.detail.thread);
+    }
+    return (starredThreads = starredThreads);
+  }
+
+  function checkIfAllStarred(
+    selection: Set<Thread>,
+    currentStarredThreads: Set<Thread>,
+  ): boolean {
+    if (
+      selection.size > currentStarredThreads.size ||
+      selection.size === 0 ||
+      currentStarredThreads.size === 0
+    ) {
+      return false;
+    }
+    for (let setThread of selection) {
+      if (!currentStarredThreads.has(setThread)) {
+        return false;
+      }
+    }
+    return true;
+  }
+
+  function starOpenedThread() {
+    if (openedEmailData !== null) {
+      if (starredThreads.has(openedEmailData)) {
+        starredThreads.delete(openedEmailData);
+        openedEmailData.starred = false;
+      } else {
+        starredThreads.add(openedEmailData);
+        openedEmailData.starred = true;
+      }
+    }
+  }
+
+  function onStarSelected(event: MouseEvent) {
+    dispatchEvent("onStarSelected", { event });
+    if (areAllSelectedStarred) {
+      selectedThreads.forEach((t) => {
+        starredThreads.delete(t);
+        t.starred = false;
+      });
+    } else {
+      selectedThreads.forEach((t) => {
+        starredThreads.add(t);
+        t.starred = true;
+      });
+    }
+    return (starredThreads = starredThreads);
+  }
+
+  function checkIfSelectionIsUnread(selectedSet: Set<Thread>): boolean {
+    for (let setThread of selectedSet) {
+      if (!setThread.unread) {
+        return false;
+      }
+    }
+    return true;
+  }
+
+  function onChangeSelectedReadStatus(event: MouseEvent) {
+    dispatchEvent("onChangeSelectedReadStatus", { event });
+    if (areAllSelectedUnread) {
+      selectedThreads.forEach((t) => {
+        t.unread = false;
+      });
+    } else {
+      selectedThreads.forEach((t) => {
+        t.unread = true;
+      });
+    }
+    return (selectedThreads = selectedThreads);
+  }
+
+  function markOpenedThreadUnread() {
+    if (openedEmailData !== null) {
+      openedEmailData.expanded = false;
+      openedEmailData.unread = true;
+      openedEmailData = null;
+    }
+    return (selectedThreads = selectedThreads);
+  }
+
+  async function onDeleteSelected(event: MouseEvent) {
+    if (openedEmailData) {
+      openedEmailData.expanded = false;
+      openedEmailData.folders = ["trash"];
+      starredThreads.delete(openedEmailData);
+      selectedThreads.delete(openedEmailData);
+      openedEmailData = null;
+    } else {
+      dispatchEvent("onDeleteSelected", { event });
+      selectedThreads.forEach((thread) => {
+        thread.folders = ["trash"];
+        starredThreads.delete(thread);
+      });
+      selectedThreads.clear();
+    }
+    return (inboxThreads = inboxThreads), (selectedThreads = selectedThreads);
```

Comment:
@AaronDDM deleting action

Reply:
So @isabellaenriquez it turns out that this might be a bit more complicated than we first anticipated. 

In order for us to change the folder of a particular thread, we need to know the `folder_id`. That is obtained via the [GET /folders](https://developer.nylas.com/docs/api/#get/folders) endpoint. We can lookup the `trash` folder ID via this endpoint (the names are standard). Once we have that ID, we have to supply it to the Thread object via `Thread.folder_id`. When we have that, we have to perform a `PUT /threads/{id}` request to actually update the API.

I recommend we do this in a separate story since it involves modifications to the middleware as well (we need to give the component access to `PUT /threads/{id}` and `GET /folders` (assuming it doesn't already have this ability).

For now just remove the following lines of code, and the rest can remain:
```      selectedThreads.forEach((thread) => {
        thread.folders = ["trash"];
        starredThreads.delete(thread);
      });
 ```

cc. @ozsivanov